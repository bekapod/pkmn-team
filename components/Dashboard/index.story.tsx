// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import Dashboard from ".";
import { Team } from "../../types";

const mockData: Team[] = [
  {
    insertedAt: "2018-06-08T21:15:14.723Z",
    id: "1",
    members: [
      {
        id: "1",
        order: 1,
        pokemon: {
          id: "1",
          name: "bulbasaur",
          pokedexId: 1,
          sprite: "1.png",
          types: [{ name: "POISON" }, { name: "GRASS" }]
        }
      },
      {
        id: "2",
        order: 2,
        pokemon: {
          id: "25",
          name: "pikachu",
          pokedexId: 25,
          sprite: "25.png",
          types: [{ name: "ELECTRIC" }]
        }
      }
    ],
    name: "Team 1"
  },
  {
    insertedAt: "2018-06-08T21:15:14.723Z",
    id: "2",
    members: [
      {
        id: "1",
        order: 1,
        pokemon: {
          id: "4",
          name: "charmander",
          pokedexId: 4,
          sprite: "4.png",
          types: [{ name: "FIRE" }]
        }
      },
      {
        id: "2",
        order: 2,
        pokemon: {
          id: "25",
          name: "pikachu",
          pokedexId: 25,
          sprite: "25.png",
          types: [{ name: "ELECTRIC" }]
        }
      },
      {
        id: "3",
        order: 3,
        pokemon: {
          id: "93",
          name: "haunter",
          pokedexId: 93,
          sprite: "93.png",
          types: [{ name: "GHOST" }, { name: "POISON" }]
        }
      }
    ],
    name: "Team 2"
  }
];

storiesOf("Dashboard", module)
  .add("with teams", () => <Dashboard teams={mockData} />)
  .add("without teams", () => <Dashboard teams={[]} />);
