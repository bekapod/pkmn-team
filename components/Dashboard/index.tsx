import Link from "next/link";
import React, { Fragment } from "react";
import { Team } from "../../types";
import CenteredRow from "../CenteredRow";
import { CtaInternalLink } from "../Cta";
import TeamCard from "../TeamCard";
import TeamGrid from "../TeamGrid";

interface Props {
  teams?: Team[];
}

const Dashboard = ({ teams }: Props): JSX.Element => (
  <Fragment>
    <CenteredRow>
      <Link href="/team/create/" passHref>
        <CtaInternalLink>Create a team</CtaInternalLink>
      </Link>
    </CenteredRow>
    {teams && (
      <TeamGrid>
        {teams.map(
          (team): JSX.Element => (
            <TeamCard key={`Team: ${team.id}`} team={team} />
          )
        )}
      </TeamGrid>
    )}
  </Fragment>
);

export default Dashboard;
