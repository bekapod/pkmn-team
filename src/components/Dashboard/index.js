// @flow
import React, { Fragment } from "react";
import TeamCard from "../TeamCard";
import Heading from "../Heading";
import SectionContainer from "../SectionContainer";
import CenteredRow from "../CenteredRow";
import TeamGrid from "../TeamGrid";
import { CtaInternalLink } from "../Cta";
import type { Team } from "../../types";

type Props = {
  teams: Array<Team>,
  loading: boolean
};

const Dashboard = ({ teams, loading }: Props) => (
  <Fragment>
    <Heading>My Teams</Heading>
    <SectionContainer>
      <CenteredRow>
        <CtaInternalLink to="/team/create/">Create a team</CtaInternalLink>
      </CenteredRow>
      {!loading && (
        <TeamGrid>
          {teams.map(team => (
            <TeamCard key={`Team: ${team.id}`} team={team} />
          ))}
        </TeamGrid>
      )}
    </SectionContainer>
  </Fragment>
);

export default Dashboard;
