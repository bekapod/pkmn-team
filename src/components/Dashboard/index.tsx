import React, { Fragment } from "react";
import { ITeam } from "../../types";
import CenteredRow from "../CenteredRow";
import { CtaInternalLink } from "../Cta";
import TeamCard from "../TeamCard";
import TeamGrid from "../TeamGrid";

interface IProps {
  teams?: ITeam[];
}

const Dashboard = ({ teams }: IProps) => (
  <Fragment>
    <CenteredRow>
      <CtaInternalLink to="/team/create/">Create a team</CtaInternalLink>
    </CenteredRow>
    {teams && (
      <TeamGrid>
        {teams.map(team => (
          <TeamCard key={`Team: ${team.id}`} team={team} />
        ))}
      </TeamGrid>
    )}
  </Fragment>
);

export default Dashboard;
