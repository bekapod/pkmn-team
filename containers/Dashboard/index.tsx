import { getOr } from "lodash/fp";
import React from "react";
import { Query } from "react-apollo";
import Dashboard from "../../components/Dashboard";
import { getAllTeams } from "../../queries/team";

const DashboardContainer = (): JSX.Element => (
  <Query query={getAllTeams}>
    {({ data /* , loading, error */ }): JSX.Element => (
      <Dashboard teams={getOr([], "teams", data)} />
    )}
  </Query>
);

export default DashboardContainer;
