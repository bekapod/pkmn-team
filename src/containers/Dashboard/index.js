// @flow
import React from "react";
import { Query } from "react-apollo";
import { getAllTeams } from "../../queries/team";
import Dashboard from "../../components/Dashboard";
import type { Team } from "../../types";

/* eslint-disable react/no-unused-prop-types */
type Response = {
  data: {
    allTeams: Array<Team>
  },
  loading: boolean
};

const DashboardContainer = () => (
  <Query query={getAllTeams}>
    {({ data, loading /* , error */ }: Response) => (
      <Dashboard teams={data.allTeams} loading={loading} />
    )}
  </Query>
);

export default DashboardContainer;
