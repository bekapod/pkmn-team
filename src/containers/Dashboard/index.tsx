import { getOr } from "lodash/fp";
import React from "react";
import { Query } from "react-apollo";
import Dashboard from "../../components/Dashboard";
import Page from "../../components/Page";
import { getAllTeams } from "../../queries/team";

const DashboardContainer = () => (
  <Query query={getAllTeams}>
    {({ data, loading, error }) => (
      <Page title="My Teams" loading={loading} error={error}>
        <Dashboard teams={getOr([], "allTeams", data)} />
      </Page>
    )}
  </Query>
);

export default DashboardContainer;
