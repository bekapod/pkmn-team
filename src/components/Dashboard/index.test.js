// @flow
import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Dashboard from ".";

const mockData = [
  {
    id: "1",
    name: "Team 1",
    createdAt: "2018-06-08T21:15:14.723Z",
    members: [
      {
        id: "1",
        pokemon: {
          name: "bulbasaur",
          pokedexId: 1,
          sprite: "1.png",
          types: ["POISON", "GRASS"]
        }
      },
      {
        id: "2",
        pokemon: {
          name: "pikachu",
          pokedexId: 25,
          sprite: "25.png",
          types: ["ELECTRIC"]
        }
      }
    ]
  },
  {
    id: "2",
    name: "Team 2",
    createdAt: "2018-06-08T21:15:14.723Z",
    members: [
      {
        id: "1",
        pokemon: {
          name: "charmander",
          pokedexId: 4,
          sprite: "4.png",
          types: ["FIRE"]
        }
      },
      {
        id: "2",
        pokemon: {
          name: "pikachu",
          pokedexId: 25,
          sprite: "25.png",
          types: ["ELECTRIC"]
        }
      },
      {
        id: "3",
        pokemon: {
          name: "haunter",
          pokedexId: 93,
          sprite: "93.png",
          types: ["GHOST", "POISON"]
        }
      }
    ]
  }
];

const incompleteMockData = [
  {
    id: "1",
    name: "Team 1",
    createdAt: "2018-06-08T21:15:14.723Z",
    members: [
      {
        id: "1",
        pokemon: {
          name: "bulbasaur",
          sprite: "1.png",
          types: ["POISON", "GRASS"]
        }
      },
      {
        id: "2",
        pokemon: {
          pokedexId: 25,
          sprite: "25.png",
          types: ["ELECTRIC"]
        }
      }
    ]
  },
  {
    id: "2",
    createdAt: "2018-06-08T21:15:14.723Z",
    members: [
      {
        id: "1"
      },
      {
        id: "2",
        pokemon: {
          name: "pikachu",
          pokedexId: 25,
          sprite: "25.png",
          types: ["ELECTRIC"]
        }
      },
      {
        id: "3",
        pokemon: {
          name: "haunter",
          pokedexId: 93,
          sprite: "93.png"
        }
      }
    ]
  }
];

describe("<Dashboard />", () => {
  it("renders all teams with associated pokemon", () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={["/"]}>
        <Dashboard teams={mockData} />
      </MemoryRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe("when dashboard is loading", () => {
    it("renders correctly", () => {
      const tree = renderer.create(
        <MemoryRouter initialEntries={["/"]}>
          <Dashboard loading />
        </MemoryRouter>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe("when dashboard has no data", () => {
    it("renders correctly", () => {
      const tree = renderer.create(
        <MemoryRouter initialEntries={["/"]}>
          <Dashboard teams={[]} />
        </MemoryRouter>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe("when dashboard as incomplete/invalid data", () => {
    it("doesn't crash", () => {
      const tree = renderer.create(
        <MemoryRouter initialEntries={["/"]}>
          <Dashboard teams={incompleteMockData} />
        </MemoryRouter>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
