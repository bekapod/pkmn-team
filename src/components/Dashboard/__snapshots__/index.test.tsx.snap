// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders all teams with associated pokemon 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Teams
  </h1>,
  <div
    className="SectionContainer-xjlkvu-0 jZbaLj"
  >
    <div
      className="CenteredRow-f3rx2f-0 hNpdbY"
    >
      <a
        className="Cta__CtaInternalLink-h8nk1r-0 jNQDYG"
        href="/team/create/"
        onClick={[Function]}
      >
        Create a team
      </a>
    </div>
    <div
      className="TeamGrid__CardGrid-sc-12cfyys-0 dpqImA"
    >
      <article
        className="Card__CardWrapper-sc-3o7tcb-0 bTjMad"
      >
        <header
          className="Card__CardHeader-sc-3o7tcb-1 cHrjuj"
        >
          <h2
            className="CardHeading-mmeo34-0 ebVIiR"
          >
            Team 1
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-3o7tcb-2 fDtTnd"
        >
          <div
            className="CardMeta__CardMetaWrapper-yqfxft-0 kTYlws"
          >
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                2
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 kdjbDA"
          >
            <img
              alt="bulbasaur sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/1.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #1 Bulbasaur
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 jHkiSl"
                    type="GRASS"
                  >
                    GRASS
                  </span>
                </li>
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 jjNEEF"
                    type="POISON"
                  >
                    POISON
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
      <article
        className="Card__CardWrapper-sc-3o7tcb-0 bTjMad"
      >
        <header
          className="Card__CardHeader-sc-3o7tcb-1 luajU"
        >
          <h2
            className="CardHeading-mmeo34-0 ebVIiR"
          >
            Team 2
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-3o7tcb-2 fDtTnd"
        >
          <div
            className="CardMeta__CardMetaWrapper-yqfxft-0 kTYlws"
          >
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                3
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-yqfxft-1 hPoDTe"
            >
              <span
                className="CardMeta__CardMetaLabel-yqfxft-2 hiNCFT"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-yqfxft-3 jqYdyo"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 pbtXk"
          >
            <img
              alt="charmander sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/4.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #4 Charmander
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 QKQcU"
                    type="FIRE"
                  >
                    FIRE
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 dAGLvI"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 byzsso"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-sc-1qiwepa-0 IOTuc"
          >
            <img
              alt="haunter sprite"
              className="PokemonLine__RowImage-sc-1qiwepa-1 hUcFnr"
              src="/sprites/93.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-sc-1qiwepa-2 edSfRl"
              >
                #93 Haunter
              </div>
              <ul
                className="InlineList-sc-1dn69qz-0 YzWxJ"
              >
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 hPYoBY"
                    type="GHOST"
                  >
                    GHOST
                  </span>
                </li>
                <li>
                  <span
                    className="TypeTag-sc-1dttu8y-0 jjNEEF"
                    type="POISON"
                  >
                    POISON
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>,
]
`;

exports[`<Dashboard /> when dashboard has no data renders correctly 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Teams
  </h1>,
  <div
    className="SectionContainer-xjlkvu-0 jZbaLj"
  >
    <div
      className="CenteredRow-f3rx2f-0 hNpdbY"
    >
      <a
        className="Cta__CtaInternalLink-h8nk1r-0 jNQDYG"
        href="/team/create/"
        onClick={[Function]}
      >
        Create a team
      </a>
    </div>
    <div
      className="TeamGrid__CardGrid-sc-12cfyys-0 dpqImA"
    />
  </div>,
]
`;

exports[`<Dashboard /> when dashboard is loading renders correctly 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Teams
  </h1>,
  <div
    className="SectionContainer-xjlkvu-0 jZbaLj"
  >
    <div
      className="CenteredRow-f3rx2f-0 hNpdbY"
    >
      <a
        className="Cta__CtaInternalLink-h8nk1r-0 jNQDYG"
        href="/team/create/"
        onClick={[Function]}
      >
        Create a team
      </a>
    </div>
  </div>,
]
`;
